src/api/AppActivity.js

```diff
@@ -0,0 +1,120 @@
+/**
+ * @flow
+ * @file Helper for the box App Activity API
+ * @author Box
+ */
+import MarkerBasedAPI from './MarkerBasedAPI';
+import { ERROR_CODE_DELETE_APP_ACTIVITY } from '../constants';
+import { APP_ACTIVITY_FIELDS_TO_FETCH } from '../utils/fields';
+
+class AppActivity extends MarkerBasedAPI {
+    /**
+     * Map an entry from the AppActivity API to an AppActivityItem.
+     * occurred_at -> created_at
+     *
+     * @param {Object} item - A single entry in the AppActivity API entiries list
+     */
+    mapAppActivityItem(item: Object): AppActivityItem {
+        const { occurred_at, ...rest } = item;
+        return {
+            created_at: occurred_at,
+            ...rest,
+        };
+    }
+
+    /**
+     * API URL for getting App Activity on a file
+     *
+     * @return {string} Url for all app activity on a file
+     */
+    getUrl(): string {
+        return `${this.getBaseApiUrl()}/app_activities`;
+    }
+
+    /**
+     * API URL for deleting app activity from a file
+     *
+     * @param id - ID of an app activity item
+     * @return {string} - URL to delete app activity
+     */
+    getDeleteUrl(id: string): string {
+        if (!id) {
+            throw new Error('Missing file id!');
+        }
+
+        return `${this.getUrl()}/${id}`;
+    }
+
+    /**
+     * Generic success handler
+     *
+     * @param {AppActivityItems} data - the response data
+     */
+    successHandler = ({ entries = [] }: AppActivityItems): void => {
+        if (this.isDestroyed() || typeof this.successCallback !== 'function') {
+            return;
+        }
+
+        const activityEntries = entries.map(this.mapAppActivityItem);
+        this.successCallback({ entries: activityEntries, total_count: activityEntries.length });
+    };
+
+    /**
+     * API for fetching App Activity on a file
+     *
+     * @param {string} id - the file id
+     * @param {Function} successCallback - the success callback
+     * @param {Function} errorCallback - the error callback
+     * @param {number} [limit] - the max number of app activity items to return.
+     * @returns {void}
+     */
+    getAppActivity(id: string, successCallback: Function, errorCallback: ElementsErrorCallback, limit?: number): void {
+        const requestData = {
+            item_id: id,
+            item_type: 'file',
+            fields: APP_ACTIVITY_FIELDS_TO_FETCH.toString(),
+        };
+
+        this.markerGet({
+            id,
+            limit,
```

Comment:
if limit is not passed in are we passing `null: null` to marketGet?

Reply:
if we're including it to be generic, then I would argue that the fields should be appended in feed where we make the call. We don't know what fields will be needed by a different caller,

Reply:
By default, `limit` is set to 1000, for`markerGet()`. Are you suggesting we just remove it, or *also* set it to 1000 inside of Feed.js?